### PR TITLE
Runtime flag: Removed envoy.reloadable_features.overload_manager_error_unknown_action

### LINKED ISSUE #31981: envoy_reloadable_features_overload_manager_error_unknown_action deprecation
Your change #27711 (Overload Manager: Validate configured overload actions.) introduced a runtime guarded feature. It has been 6 months since the new code has been exercised by default, so it's time to remove the old code path. This issue tracks source code cleanup so we don't forget.

### PR DESCRIPTION
**Pull request by KBaichoo**
and legacy codepaths.

Commit Message: Removed runtime flag envoy.reloadable_features.overload_manager_error_unknown_action and and legacy codepaths.
Additional Description:
Risk Level: low
Testing: 
Docs Changes: included changelog
Release Notes: included
Platform Specific Features:
[Optional Runtime guard:]
Fixes #31981

### CODE REVIEW

**Review by adisuissa — DISMISSED** [review]
Thanks for cleaning this up, LGTM!
Left a small nit.

/assign-from @envoyproxy/senior-maintainers

**Review by alyssawilk — APPROVED** [review]
LGTM modulo outstanding comments

**Review by nezdolik — APPROVED** [review]

File: source/server/overload_manager_impl.cc [hunk 1]
```diff
@@ -420,13 +420,10 @@ OverloadManagerImpl::OverloadManagerImpl(Event::Dispatcher& dispatcher, Stats::S
...
 
     // Validate that this is a well known overload action.
-    if (Runtime::runtimeFeatureEnabled(
-            "envoy.reloadable_features.overload_manager_error_unknown_action")) {
-      auto& well_known_actions = OverloadActionNames::get().WellKnownActions;
-      if (std::find(well_known_actions.begin(), well_known_actions.end(), name) ==
-          well_known_actions.end()) {
-        throw EnvoyException(absl::StrCat("Unknown Overload Manager Action ", name));
-      }
+    auto& well_known_actions = OverloadActionNames::get().WellKnownActions;
```
[the remark this answers is not in the document]
Comment (KBaichoo) [reply source/server/overload_manager_impl.cc]: done.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**